From 839d6ba7ea6838f510cfe806cec1f74b07059aef Mon Sep 17 00:00:00 2001
From: Tristan <tristan@mangadex.org>
Date: Thu, 19 Oct 2023 03:43:01 +0100
Subject: [PATCH] MINOR: lua: do not forward logs to stderr if a logger is
 configured

After making it configurable in commit edf197f105 ("MINOR: lua: Add flags
to configure logging behaviour"), this patch changes the default value
of tune.lua.log-stderr from 'on' (unconditionally forward LUA logs to
stderr) to 'auto' (only forward LUA logs to stderr if logging via a
standard logger is disabled, or none is configured)

Since this is a change in behaviour, it shouldn't be backported
---
 doc/configuration.txt | 2 +-
 doc/lua.txt           | 4 ++--
 src/hlua.c            | 2 +-
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/doc/configuration.txt b/doc/configuration.txt
index d620ea554..b9731daa4 100644
--- a/doc/configuration.txt
+++ b/doc/configuration.txt
@@ -3198,7 +3198,7 @@ tune.lua.log-stderr { on | auto | off }
   Please note that, when enabled, this logging is in addition to the logging
   configured via tune.lua.log.
 
-  Defaults to 'on'.
+  Defaults to 'auto'.
 
 tune.max-checks-per-thread <number>
   Sets the number of active checks per thread above which a thread will
diff --git a/doc/lua.txt b/doc/lua.txt
index 5516ac36a..08baf6035 100644
--- a/doc/lua.txt
+++ b/doc/lua.txt
@@ -631,8 +631,8 @@ It displays a log during the HAProxy startup:
    [alert] 285/083533 (14465) : Hello World !
 
 Note: By default, logs originating from a LUA script are sent to the loggers
-applicable to the current proxy, if any, and additionally to stderr. See
-tune.lua.log and tune.lua.log-stderr for more.
+applicable to the current proxy, if any. If none are configured, they are
+instead sent to stderr. See tune.lua.log and tune.lua.log-stderr for more.
 
 Default path and libraries
 --------------------------
diff --git a/src/hlua.c b/src/hlua.c
index b9d9bb0f1..8a14ed501 100644
--- a/src/hlua.c
+++ b/src/hlua.c
@@ -78,7 +78,7 @@
 #define HLUA_TUNE_LOG_STDERR_MASK 0x00000070
 
 /* default flag values */
-#define HLUA_TUNE_FLAGS_DFLT	  0x00000011
+#define HLUA_TUNE_FLAGS_DFLT	  0x00000021
 
 /* flags made of HLUA_TUNE_* */
 static uint hlua_tune_flags = HLUA_TUNE_FLAGS_DFLT;
-- 
2.41.0

